### ISSUE TITLE
Auth issue when using own api in sveltekit

### ISSUE THREAD

**Issue by 24jr** [supabase/auth-helpers#531]
When calling api with fetch in +page.server.js it screws up the client-side auth session when I call my own api. Other apis work fine

Go to /test page to see the difference when commenting out different fetch calls in /routes/test/+page.server.js

I followed docs here https://supabase.com/docs/guides/auth/auth-helpers/sveltekit
I've mentioned on discord before and others have similar issue https://discord.com/channels/839993398554656828/1081629955223453706 

https://github.com/24jr/supaauthtest

**Comment by silentworks** [comment]
@24jr your demo is incomplete as there is no way to sign in since there is no route for `/auth` which is where your magic link signup action is supposed to be located. Please provide a complete example and I'll take a look again.

**Comment by 24jr** [comment]
@silentworks You're right! I was copying the most minimally needed code over and missed that file. I just added it and I think should be functional now to show you issue

**Comment by silentworks** [comment]
I'm not getting the issue you mentioned on Discord however there is an odd behaviour of the cookie's httpOnly state changing when calling an internal endpoint.

**Comment by 24jr** [comment]
@silentworks Yeah i noticed odd crashes from cookie stuff recently as well. I just noticed they depricated otp types magiclink and signup, which made it complicated for me to to verify...whatever not important... but I just changed that to new otp name email, which may help.

To be clear if you refresh the page `http://localhost:5173/test` you are logged in? the top right will show a login button if you are not logged in and the profile icon if you are logged in. For me a page refresh looks like this, which is not logged in and never shows login on refresh.

<img width="734" alt="Screen Shot 2023-05-05 at 10 50 54 AM" src="https://user-images.githubusercontent.com/34662182/236492280-6b866345-e7f0-432a-b17d-1af6c8556cd1.png">

**Comment by silentworks** [comment]
I refreshed and I was still logged in. I'm not able to replicate the behavior you mention in this issue.

**Comment by david-plugge** [comment]
It seems to be an issue with sveltekit itself, i created an issue over there https://github.com/sveltejs/kit/issues/9901

**Comment by 24jr** [comment]
For now i put the project in dev site. Idk if that makes it easier to see the issue in a live env before getting too hands-on. https://github.com/24jr/supaauthtest signing in and refreshing on /test page will wrongly not display signed in.

**Comment by 24jr** [comment]
"@sveltejs/kit": "^1.18.0", seems to have fixed this on reload though onAuthStateChange isn't triggering on signin but works on page reload. The fix just came out though so ill mess with it
